app/src/main/java/org/fossasia/openevent/app/core/event/list/EventsListAdapter.java

```diff
@@ -13,24 +13,56 @@
 import org.fossasia.openevent.app.databinding.EventLayoutBinding;
 import org.fossasia.openevent.app.databinding.HeaderLayoutBinding;
 import org.fossasia.openevent.app.ui.HeaderViewHolder;
+import org.fossasia.openevent.app.utils.service.DateService;
 
+import java.text.ParseException;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 
+import timber.log.Timber;
+
 class EventsListAdapter extends RecyclerView.Adapter<EventsListAdapter.EventRecyclerViewHolder>
     implements StickyRecyclerHeadersAdapter<HeaderViewHolder> {
 
     private final List<Event> events;
+
     private final Bus bus;
     private final EventsPresenter eventsPresenter;
     private boolean sortByName;
 
+
+    private final List<Event> liveEvents = new ArrayList<>();
+    private final List<Event> pastEvents = new ArrayList<>();
+    private final List<Event> upcomingEvents = new ArrayList<>();
+    private final List<Event> selectedEvents = new ArrayList<>();
+
     EventsListAdapter(List<Event> events, Bus bus, EventsPresenter eventsPresenter) {
         this.events = events;
         this.bus = bus;
         this.eventsPresenter = eventsPresenter;
     }
 
+    public void categorizeEvents() {
+        liveEvents.clear();
+        pastEvents.clear();
+        upcomingEvents.clear();
+        for (Event event : events) {
+            try {
+                String category = DateService.getEventStatus(event);
+                if ("past".equalsIgnoreCase(category))
+                    pastEvents.add(event);
+                else if ("live".equalsIgnoreCase(category))
+                    liveEvents.add(event);
+                else
+                    upcomingEvents.add(event);
+            } catch (ParseException e) {
+                Timber.e(e);
+            }
+        }
+        notifyDataSetChanged();
```

Comment:
Implement this through filter instead

Reply:
Sorry, I didn't get this one. AFAIK filter is used to filter items emitted by Observable, but here we want every item to be updated. So why use filter? 
Could you please elaborate a little?
Also I was thinking of moving the methods added to the adapter, to the presenter. In that case we can just update the `events` List every time. But I am stuck on how to listen for its changes.

Reply:
I meant use the adapter filtering feature instead of maintaining 3 lists